mappings/clacso.py

```diff
@@ -1,23 +1,108 @@
+
+    
+    def __init__(self, source, namespace, constants):
+        super(CLACSOMapping, self).__init__(source, namespace, constants)
         self.mapping = self.createMapping()
     
     def createMapping(self):
         return {
-            'title': ('title', '{0}'),
-            'authors': ('authors', '{0}|||true'),
-            'abstract': ('description', '{0}'),
-            'dates': [('publication_date', '{0}|publication_date')],
-            'identifiers': 
-                [('isbn', '{0}|isbn')]
-            ,
-            'spatial': ('edition_location', '{0}'),
-            'has_part': [('url', '1|{0}|clacso|application/pdf|{{"catalog": false, "download": true, "reader": false, "embed": false}}')] 
+            'title': ('./dc:title/text()', '{0}'),
+            'authors': [('./dc:creator/text()', '{0}|||true')],
+            'abstract': ('./dc:description/text()', '{0}'),
+            'dates': [
+                (
+                    [
+                        './dc:date/text()',
+                    ],
+                    '{0}'
+                )
+            ],
+            'identifiers': [
+                ('./dc:identifier/text()', '{0}'),
+            ],
+            'medium' : [
+                ('./dc:type/text()', '{0}')
+            ],
+            'has_part': [(
+                './dc:identifier/text()',
+                '1|{0}|clacso|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
+            )]
         }
 
     def applyFormatting(self):
         self.record.source = 'clacso'
-        self.record.source_id = f'{self.record.source}'
+        if self.record.identifiers:
+            self.format_identifiers(self.record.identifiers)
+        print(self.record.identifiers)
+        self.format_source_id()
+        self.format_medium()
+        self.format_dates()
+        self.format_has_part()
+
+    def format_source_id(self):
+        for identifier in self.record.identifiers:
+            if self.HANDLE_URL in identifier:
+                id = identifier.split('/')[-1]
+                self.record.source_id = f'{self.record.source}|{id}'
+                break
+
+    def format_medium(self):
+        if self.record.medium:
+            type_list = ['book', 'bookpart', 'part', 'chapter', 'bibliography', 'appendix', 'index',
+                    'foreword', 'afterword', 'review', 'article', 'introduction']
+            for medium in self.record.medium:
+                if '/' in medium:
+                    medium_value = medium.split('/')[-1]
+                    if medium_value.lower() in type_list:
+                        self.record.medium = medium_value.lower()
+                        break
+                elif '/' not in medium and medium.lower() in type_list:
+                    medium_value = medium
+                    self.record.medium = medium_value.lower()
+                    break
+        else:
+            self.record.medium = ''
+
+    def format_dates(self):
```

Comment:
You can use list comprehension to make this method cleaner (as you did with clean_has_part below)

Reply:
I appreciate the suggestion Jackie.